server/src/main/java/org/apache/druid/server/http/DataSourcesResource.java

```diff
@@ -596,6 +661,44 @@ private Response getServedSegmentsInInterval(
     }
   }
 
+  private Response getUnusedSegmentsInInterval(
+      Iterator<DataSegment> unusedDataSegments,
+      boolean full,
+      Predicate<Interval> intervalFilter,
+      final Comparator<Interval> comparator
+  )
+  {
+    if (full) {
```

Comment:
IMO, it'd be better to break these into two separate small functions:
1. `getUnusedSegmentsInInterval`
2. `getUnusedSegmentsStatsInInterval`

Reply:
fixed